internal/snmp/translate_test.go

```diff
@@ -87,3 +89,60 @@ func TestTrapLookupFail(t *testing.T) {
 		})
 	}
 }
+
+type TestingMibLoader struct {
+	folders []string
+	files   []string
+}
+
+func (t *TestingMibLoader) appendPath(path string) {
+	t.folders = append(t.folders, path)
+}
+
+func (t *TestingMibLoader) loadModule(path string) error {
+	t.files = append(t.files, path)
+	return nil
+}
+func TestFolderLookup(t *testing.T) {
+	var folders []string
+	var givenPath []string
+
+	tests := []struct {
+		name         string
+		mibPath      [][]string
+		paths        [][]string
+		files        []string
+		windowsFiles []string
+	}{
+		{
+			name:         "loading folders",
+			mibPath:      [][]string{{"testdata", "loadMibsFromPath"}},
+			paths:        [][]string{{"testdata", "loadMibsFromPath"}, {"testdata", "loadMibsFromPath", "linkTarget"}, {"testdata", "loadMibsFromPath", "root"}},
+			files:        []string{"emptyFile"},
+			windowsFiles: []string{"emptyFile", "symlink"},
+		},
```

Comment:
Add another case for a directory structure that is 2-3 deep:

testdata
-- loadMibsFromPath
---- root
------ file here

and point the test at root.